### ISSUE TITLE
Trying to get the boat to move (plugins:BoatMarker)

### ISSUE THREAD

**Issue by ZerNox** [python-visualization/folium#718]
Ok im really new here guys,  but i can´t make the animation for the BoatMarker to work. 
I dont get any errors setting speed but, the boats are stuck.

```python
import folium
import os 
from folium import plugins 

m = folium.Map([30, 0], zoom_start=3)

plugins.BoatMarker(
    location=(34, -43),
    heading=45,
    wind_heading=150,
    speed=10,
    wind_speed=45,
    color='#8f8'
).add_to(m)

plugins.BoatMarker(
    location=(46, -30),
    heading=-20,
    speed=100,
    wind_heading=46,
    wind_speed=25,
    color='#88f'
).add_to(m)


#m.save(os.path.join('results', 'Plugins_3.html'))

m
```

**Comment by ocefpaf** [comment]
Looks like a bug. We need to check if the boat plugin works with leaflet 1.20 or if needs an update.

PRs welcome :wink:

**Comment by ZerNox** [comment]
Ocefpaf, is that an upcoming release after 1.2.0? 
Tried this on 0.5.0+40.g3f9e612 
Im not sure how to update to Latest leaflet version to try it out. 

But i guess that it have something to do with 
`_default_js = [
    ('leaflet',
     'https://cdn.jsdelivr.net/npm/leaflet@1.2.0/dist/leaflet.js'),`

**Comment by ocefpaf** [comment]
Latest release is already using leaflet `1.2.0`, see https://github.com/python-visualization/folium/blob/v0.5.0/folium/map.py#L33

I believe that the `BoatMarker` plugin is not compatible with it, but I did not check [upstream](https://github.com/thomasbrueggemann/leaflet.boatmarker) though.

I am in the process of adding an option to use other leaflet version in the map, that will make it easier to debug this.

**Comment by psychemedia** [comment]
Is this feature still under development, or has it been superceded by other plugins?

I'm looking for a really simple plugin that will let me take a dataframe containing `lat`, `lon` and `timestamp` columns and render it as a moving marker, either with or without a t(r)ail (more complex would be a trail that just shows a trail over the last N points of the line, or over the previous within a specified time window).

It could also be handy to support several animated points, eg via a dataframe containing `lat`, `lon`, `timestamp` and `group` columns, rendered as separate moving markers, either with or without distinctly coloured t(r)ails.

**Comment by ocefpaf** [comment]
> Is this feature still under development, or has it been superceded by other plugins?

I don't know. One would need to check the plugin upstream and see if that is still under development. As we move forward with `folium` we are removing plugins and recommending people to create independent package-wrappers for the plugins they want to use. This will reduce maintenance burden and will allow for a plugin to evolve faster.

TL;DR the BoatMarker is a good candidate for an external package.

**Comment by Conengmo** [comment]
Looking at https://github.com/thomasbrueggemann/leaflet.boatmarker, it seems the library itself doesn't provide animation. The demo actually contains some custom JS to change the heading of the boat.

If we wanted to have this in folium, we would have to write that ourselves. Or potentially combine it with some timedimension library.

I'll close the issue since it's out of scope for folium. If somebody wants to make it, great! Please share it here. We can either include it in this repo if the maintenance burden is limited, or you can host the repo yourself and we will link to it.
